### PR TITLE
remove debug console output

### LINKED ISSUE #334: bin/kuzzle plugins --get returns [Object] for embeded properties
```
kuzzle # ./bin/kuzzle plugins --get kuzzle-plugin-logger                                                                                                                        /var/app (kuz-536-internal-state-updates-hooks) 8d257f306019
{ npmVersion: '2.0.2',
  activated: true,
  config: 
   { services: { file: [Object], stdout: [Object] },
     loadedBy: 'all' } }
```

We should get the full plugin configuration.

### PR DESCRIPTION
**Pull request by benoitvidis**
## Changelog

* options `--noColors` added for all Kuzzle commands in case the end user terminal does not support ANSI colors
* Plugin configurations are now displayed with an infinite depth (fixes #334)

### CODE REVIEW

**Comment by scottinet** [comment]
+1

**Comment by jenow** [comment]
Maybe we should console.dir instead of console.log because in case of a --get we only see [Object]

**Comment by benoitvidis** [comment]
@jenow I agree but it should be handled in a new PR.
I opened the issue #334, which is under progress.

**Comment by benoitvidis** [comment]
@jenow My bad, I had not realized this console output was also used by `--list` and `--get` options.

**Comment by xbill82** [comment]
:+1:

File: bin/commands/createFirstAdmin.js [hunk 1]
```diff
@@ -230,11 +230,22 @@ var checkIfFistAdminNeeded = () => {
 
 process.stdin.setEncoding('utf8');
 
-module.exports = {
-  check: checkIfFistAdminNeeded,
-  action: () => {
-
-    checkIfFistAdminNeeded()
+module.exports = function FirstAdmin (options, run) {
+  if (!(this instanceof FirstAdmin)) {
```
Comment (ballinette) [inline bin/commands/createFirstAdmin.js]: I don't understand the usage of this statement?

File: lib/api/controllers/remoteActions/managePlugins.js [hunk 2]
```diff
@@ -9,21 +9,22 @@ var
...
+  clcError = clc.red;
 
 module.exports = function pluginsManager(plugin, options) {
   var
     kuzzleConfiguration = require('../../../config')(defaultConfig),
     dbService = new DatabaseService({config: kuzzleConfiguration}),
     deferred = q.defer();
-
+  
+  if (options.parent && options.parent.noColors) {
```
Comment (ballinette) [inline lib/api/controllers/remoteActions/managePlugins.js]: Why not writing colors options as you made in `bin/commands/*` scripts ?
```
var
  clcOk = string => options.parent.noColors ? string : clc.geen.bold(string),
  clcNotice = string => options.parent.noColors ? string : clc.cyan(string),
  clcError = string => options.parent.noColors ? string : clc.red(string);
```

Comment (benoitvidis) [inline lib/api/controllers/remoteActions/managePlugins.js]: These functions are used outside of module.exports, where the options variable does not exist.
It was much easier, notably regarding the tests we have, to keep the default behavior and override the methods only when asked.

File: bin/commands/createFirstAdmin.js (hunk 1 above)
Comment (ballinette) [inline bin/commands/createFirstAdmin.js]: ok

File: lib/api/controllers/remoteActions/managePlugins.js (hunk 2 above)
Comment (ballinette) [inline lib/api/controllers/remoteActions/managePlugins.js]: ok

**Comment by ballinette** [comment]
+1

**Pull request merged**